tmt/checks/coredump.py

```diff
@@ -0,0 +1,350 @@
+            ).stdout
+
+        if not output:
+            logger.debug("Failed to read timestamp file")
+            return ResultOutcome.ERROR
+
+        start_time = output.strip()
+
+        if self._check_for_crashes(
+                invocation.guest, logger, str(invocation.check_files_path), start_time
+                ):
+            return ResultOutcome.FAIL
+
+        return ResultOutcome.PASS
+
+
+@provides_check("coredump")
+class Coredump(CheckPlugin[CoredumpCheck]):
+    """
+    Check for system crashes using coredump.
+
+    The check monitors for any crashes caught by systemd-coredump during test
+    execution. This includes segmentation faults and other crashes that produce
+    core dumps. By default, any crash will cause the test to fail.
+
+    Example config with optional ignore patterns:
+
+    .. code-block:: yaml
+
+        check:
+          - how: coredump
+            ignore-patterns:
+              - 'Process.*\\(sleep\\).*dumped core'  # Ignore sleep crashes
+              - 'Package: ddcutil/2.1.2-2.fc41'      # Ignore dumps of a specific package
+
+    The patterns are matched against the full coredumpctl info output, which includes
+    fields like Process, Command Line, Signal, etc. You can use 'coredumpctl info'
+    to see the available fields and their format.
+
+    .. versionadded:: 1.41
+    """
+
+    _check_class = CoredumpCheck
+
+    @classmethod
+    def essential_requires(
+            cls, guest: "Guest", test: "tmt.base.Test", logger: tmt.log.Logger
+            ) -> list["tmt.base.DependencySimple"]:
+        # Avoid circular imports
+        import tmt.base
+
+        # Only require systemd-udev if we have necessary capabilities
+        required_capabilities: list[GuestCapability] = [
+            GuestCapability.CAP_SYS_ADMIN,    # Configure coredump
+            GuestCapability.CAP_SYSLOG,       # Read journal
+            ]
+
+        if not guest.facts.has_systemd and not guest.facts.has_capabilities(required_capabilities):
+            logger.debug("Missing required capabilities, skipping coredump dependencies")
+            return []
+
+        return [tmt.base.DependencySimple("systemd-udev")]  # includes coredumpctl
+
+    @classmethod
+    def before_test(
+            cls,
+            *,
+            check: "CoredumpCheck",
+            invocation: "TestInvocation",
+            environment: Optional[tmt.utils.Environment] = None,
+            logger: tmt.log.Logger,
+            ) -> list[CheckResult]:
+        """Check for crashes before the test starts."""
+        # Required capabilities for full coredump functionality
+        required_capabilities = [
+            GuestCapability.CAP_SYS_ADMIN,    # Configure coredump
+            GuestCapability.CAP_SYSLOG,       # Read journal
+            GuestCapability.CAP_DAC_READ_SEARCH,  # Access coredump files
+            GuestCapability.CAP_DAC_OVERRIDE  # Access protected files
```

Comment:
Why aren't these two capabilities tested in `essential_requires()`?

Reply:
Good question - basically this is all a guesswork, I don't actually know if we need these, or any for that matter. Also, I don't think you need `CAP_SYSLOG`, to read journal, at least on my laptop, I have cap_syslog 'not permitted', but can do journalctl, as well as run coredumpctl.

I was experimenting with some ways to figure out if want to skip the coredump check, like
 - doesn't have systemd -> skip
 - doesn't have coredumpctl
   - doesn't have permission to modify coredumpctl config -> skip
   - install coredumpctl
- etc...

But I think I'm overthinking it, imho it can be just has_systemd and can modify the config. Skip everything else, wdyt? (removing essential_requires alltogether)

Reply:
simplified in c0c05df176604cf7fa3ec96d92f952e03b886726